### ISSUE TITLE
Model Results

### ISSUE THREAD

**Issue by calimandaniel** [WangStephen/DeepCA#4]
Hello, 

Congrats for you work! I tried to reproduce the results, I divided the ImageCAS dataset into only using the RCA (using connected components 3d) and I used the data_simulation.py file to preprocess the dataset. I managed to make the model train, however, there is something wrong with the results. Sometimes when I train I have very bad results, sometimes results that are in the  right direction (still not good). Is there anything else that we should have here in order to train the model and have the correct results?

Input:

![Image](https://github.com/user-attachments/assets/882c327e-4dc8-4eac-9f5d-0cc23580fff7)

BP:

![Image](https://github.com/user-attachments/assets/678746d1-2bc6-44b1-a991-298d01082da0)

Output:

![Image](https://github.com/user-attachments/assets/e9491f64-3df9-4775-9f23-d4d301d94a03)

Best regards,
Daniel

**Comment by WangStephen** [comment]
Hi thank you so much for the trial.

Could I please ask several questions so that I can be clearer about your process:

1. Have you checked the RCA data after the split? Because not all the 1000 data can be successfully separated into RCA and LAD, I ignore these data which cannot be separated. Moreover, after separation, have you checked every data to ensure they are all RCA because not all the first connected components are in the same category, i.e. RCA.

2. For data simulation, would you please share an example of the two projections you generated?

3. Have you changed any hyper-parameters?

**Comment by calimandaniel** [comment]
Hello, 

Sure thing.

1. I took a look empirically at some examples from the dataset, they were RCA. It didn't seem nothing wrong with them.
2. I'll put some projections generated.
3. No, I didn't change any parameters.

Thank you!

<img width="349" height="352" alt="Image" src="https://github.com/user-attachments/assets/a0bc3faf-e8be-41ec-b581-ff34e8e5f486" />

<img width="359" height="363" alt="Image" src="https://github.com/user-attachments/assets/35c5fb6c-6119-410a-b248-e1eaf3039d38" />

<img width="405" height="389" alt="Image" src="https://github.com/user-attachments/assets/5d6b8f83-6c42-4982-8ed9-78d1f5e0ac02" />

<img width="335" height="330" alt="Image" src="https://github.com/user-attachments/assets/1073af73-1bdd-47aa-a892-efbb8a48f28d" />

<img width="291" height="279" alt="Image" src="https://github.com/user-attachments/assets/78eef590-40f2-47d3-ac9a-f136f2dce873" />

<img width="430" height="383" alt="Image" src="https://github.com/user-attachments/assets/110d23d4-a774-4193-bbb9-ec53bd010501" />

**Comment by WangStephen** [comment]
Hi,

1. The first two projections you show are not RCA but LAD. I suggest you fully check the data you have for training. Moreover, not all the data in the ImageCAS dataset are perfectly segmented, so you should also drop these data when checking.

2. You should generate two projections kinda like the projections I show in the Fig. 1 of the paper, which are what the RCA projections really look like in real clinics.

**Comment by calimandaniel** [comment]
Hello,

I took a look at the input data, and I removed the ones that we're not RCA. I trained the model but haven't got the results described in the paper (only Ot(1) = 42 Ot(2) = 60) . I have some questions:

1. **Sparse Inputs:** In some examples, the voxel points in the input look quite sparse. Did you keep such cases in your dataset, or did you filter them out?

<img width="1194" height="567" alt="Image" src="https://github.com/user-attachments/assets/4dd447d5-ac90-46fd-a0ce-dd7da9da0d6c" />

2. In Table 3, you describe the angles as: First projection: Primary Angle 18–42°, Secondary Angle –8–8° and Second projection: Primary Angle –8–8°, Secondary Angle 18–42°. Would the algorithm work with other angle combinations, or is retraining required for different angle types?

3. **Geometry Parameters:** For real 2D ICA preprocessing, how critical is it to match the exact DSD/DSO ranges from Table 3 (970–1010mm, 745–785mm)? Can we use the actual DICOM geometry, or should we rescale/normalize to match the synthetic data?

4. **Motion Simulation**: Since our real DICOM data doesn't have the simulated motion you applied to the second projection, should we artificially add random rotations (±10°) and translations (±8mm) to match the training data?

Here are some input examples for reference:

<img width="571" height="564" alt="Image" src="https://github.com/user-attachments/assets/5075dd76-c634-41b1-840e-65adb1e9d504" />

<img width="552" height="562" alt="Image" src="https://github.com/user-attachments/assets/bff3614b-64f5-4b50-8be5-9fb5f6dca413" />

<img width="536" height="479" alt="Image" src="https://github.com/user-attachments/assets/66d0001d-456b-4895-8da0-b44f3fc99237" />


Thank you!

**Comment by WangStephen** [comment]
Hi,

1. I don't keep all the data. I filter some data according to their 2D projections. For example, I will project them onto 2D plane using clinical geometry and if the 2D projections are not good compared to real data, I will drop them. (I plan to upload my pre-processed datasets by the end of this month at the earliest if possible, then you can try mine if you still cannot figure it out.)

2. The angles and other projection geometry parameters in simulation are based on what usually doctors will choose. If your real data are within this range, it should be fine, or you have to set your own clinical settings for the projection geometry in your settings.

3. Same as 2. This depends on your data settings. My real data are originally within this range, so no other scaling/normalisation is required and I directly use the actual clinical parameters.

4. No, you shouldn't apply motion to real data as real data originally contain motion between projections. From the real data you present here, they look similar to my cases where the first projection and second projection are not aligned (non-simultaneously acquisition resulting in motion). My proposed pipeline is to simulate such motion in CCTA data which are originally static so that after training, the model can be directly applied to the real data where there are motion already existed.

Hope these could help.

**Comment by WangStephen** [comment]
Hi @calimandaniel ,

Here is the [link](https://drive.google.com/file/d/1e4sS2E5LTc9GtKCx-WVVWQv65NipN4kC/view?usp=sharing) to the datasets I generated from the CCTA data used for training DeepCA.

Hope this helps.

regards

**Comment by calimandaniel** [comment]
Hello, 

Thanks for sharing the dataset. It's really useful. In the meanwhile I managed to curate the dataset by removing the sparse elements, achieving similar performances as described in the paper.

However, when trying to use the algorithm for the real ICA projections, something doesn't work. In the paper, is described that you use binary segmentations from 2/3 projections to do the backprojection. I used the parameters of Here I have a few questions:

1. When you have multiple projections, do you use them all to create the backprojection? Or only the ones that satisfy the angle requirements? Or maybe the 2 projections that have opposite angles? 
2. In the paper, the backprojection has values in the inverval [0, 2] (because we have 2 projections). Whenever we have >2 projections, the combined backprojection would have values in interval [0, no. of projections]? Or do you clip it into the interval [0,2]?

Here I attach one example with my projections, angles backprojection and inference result:

<img width="752" height="787" alt="Image" src="https://github.com/user-attachments/assets/84174944-b5a9-4f08-8e7c-336a309a6bbe" />

<img width="755" height="790" alt="Image" src="https://github.com/user-attachments/assets/b2ee2aa6-35ae-44b3-aeda-cb13af9df0f8" />

<img width="765" height="787" alt="Image" src="https://github.com/user-attachments/assets/2840bf72-c636-4e12-800d-33fdf163fb42" />

<img width="752" height="786" alt="Image" src="https://github.com/user-attachments/assets/88c9f41c-07a0-437e-9d40-f87cf3d7abc7" />

<img width="765" height="792" alt="Image" src="https://github.com/user-attachments/assets/bd58a79d-9e28-4730-bd02-4433c8f66e64" />

For the backprojection I also tried to use only 2 projections (angles from most opposite views) 

<img width="847" height="719" alt="Image" src="https://github.com/user-attachments/assets/5b54dc72-4bc2-4b7a-bd2a-8a5bb242ffbb" />

Result (binarized > 0.5 after model output):

<img width="603" height="584" alt="Image" src="https://github.com/user-attachments/assets/a88aae2a-dc78-4386-90b4-2389990ae4e4" />


Another example:

<img width="762" height="782" alt="Image" src="https://github.com/user-attachments/assets/e60179c9-ab9d-4683-876b-94452b5933cf" />

<img width="757" height="784" alt="Image" src="https://github.com/user-attachments/assets/a5f205cf-8ae3-42d0-9810-bea539cfb344" />

Backprojection:

<img width="806" height="687" alt="Image" src="https://github.com/user-attachments/assets/62e24048-1fda-4ae5-9ad7-3c5ea7027ed8" />

Result: 

<img width="870" height="874" alt="Image" src="https://github.com/user-attachments/assets/91c6edc4-3f3f-46a0-ad5f-3cc4a133c6ce" />

Do you know why would this be the case? 

Thank you for your kind help,
Daniel

**Comment by WangStephen** [comment]
Hi,

1. 
In the paper, any input contains backprojection from **only two** projections. The extra/third projection is only used for measurement on the 2D projection plane. 
These two real ICA projections used for creating the input cannot be random views but should follow the range (see Table 3) that you used for simulating the training datasets. (I think it can slightly be out of the range, but I have not tested this.) So the two projections should be LAO and AP Cranial views in this case. 
Therefore, if you have your own clinical settings such as different views from this paper, you can simulate your own training datasets following your clinical settings which then can work on your real ICA data.

2. 
My input values are either 0, or 1, or 2, meaning empty background, projection line from one view, and intersection point from two projections. I have also tested that making both values 1 and 2 be 1, and the model still can achieve similar results.
So no matter what values are in your real data input, you should make them consistent with your training input data. You can customise your own cases as long as you make them consistent.

Hope this can help.

**Comment by calimandaniel** [comment]
Hello Stephen,

Thanks for clarifying, that makes sense now. I’ll try to implement what you explained.

I really appreciate your guidance and assistance.

Best regards,
Daniel
